app/services/get_application_forms_waiting_for_provider_decision.rb

```diff
@@ -0,0 +1,12 @@
+class GetApplicationFormsWaitingForProviderDecision
+  def self.call
+    ApplicationChoice
+      .where(status: :awaiting_provider_decision)
+      .where('reject_by_default_at < ?', chase_provider_time_limit)
+      .where.not(id: ChaserSent.provider_decision_request.select(:chased_id))
+  end
+
+  def self.chase_provider_time_limit
+    TimeLimitCalculator.new(rule: :chase_provider_by, effective_date: Time.zone.now).call.second
```

Comment:
`effective_date` here should really be the time at which the clock started (because in theory at least the time limits might be different now). In other words counting back from the RBD date _may_ give different results to counting forward.

I'm not sure what to suggest here because we don't record the time at which the clock started (there is no `ApplicationChoice#sent_to_provider_at`) and I know you've already chatted with @tijmenb about it. I think perhaps this is the best solution we can have as things stand.

An alternative would be to drop the configurable limit (in `TimeLimitConfig`) and just count back `ApplicationChoice. reject_by_default_days / 2` days from `ApplicationChoice. reject_by_default_at`.

Reply:
Yes, you are right, the problem with `ApplicationChoice. reject_by_default_days / 2` assume the business rule will be always halfway rough RBD. We agreed with @tijmenb that we won't record a chase_provider_at date but we count backwards from the `reject_by_default_at`. I think as long as this rule changes following the current logic it should be fine. I'm more concerned about the naming and readability.

Reply:
I still haven't wrapped my head around it. Let's make sure we did the right thing though, how do we test that?

Reply:
I missed your comment @tijmenb but I could change the tests to something like
 ``` it 'returns an application that has not more than 20 working days till RBD date' do
    application_choice = create_application(
      status: 'awaiting_provider_decision',
      reject_by_default_at: TimeLimitConfig.limits_for(:chase_provider_by).first.limit.business_days.from_now,
    )
    expect(described_class.call).to include application_choice
  end```
In this case, you can change the time limit to any number and it would find the application choices that have the number of days before RBD. It would be also good to decouple the 20 days rule from the unit tests, so the rule can change and not breaking the current test. Let me know what you think.